core/src/test/java/com/capitalone/dashboard/repository/FeatureBranchRepositoryTests.java

```diff
@@ -0,0 +1,225 @@
+		featureBranchThree.setCommitIdThatTriggeredBuild("G033_TB");
+		featureBranchThree.setGitRepoUrl("api.github.com/repo/commits/G003-G033");
+		featureBranchThree.setFirstCommitTimeStamp(1499553944L);	// 2017/07/08 22:45:54
+		featureBranchThree.setDeployTimeStamp(1499653944L); 		//2017/07/10 02:32:24
+	}
+
+	@After
+	public void tearDown() {
+		featureBranchOne = null;
+		featureBranchTwo = null;
+		featureBranchThree = null;
+		featureBranchRepository.deleteAll();
+	}
+
+	@Test
+	public void validateSave() {
+		featureBranchRepository.save(featureBranchOne);
+		featureBranchRepository.save(featureBranchTwo);
+		featureBranchRepository.save(featureBranchThree);
+
+		assertTrue("Save function validation for the FeatureBranchRepository has failed",
+				featureBranchRepository.findAll().iterator().hasNext());
+	}
+
+
+	@Test
+	public void testFindById()
+	{
+		featureBranchRepository.save(featureBranchOne);
+
+		FeatureBranch testCompare = featureBranchRepository.findById(featureBranchOne.getId());
+
+		assertEquals("FeatureBranch findById() not functioning as expected",
+			testCompare.getCommitIdFirstCommit(), featureBranchOne.getCommitIdFirstCommit());
+	}
+
+
+	@Test
+	public void testFindByCommitIdThatTriggeredBuild()
+	{
+		featureBranchRepository.save(featureBranchOne);
+		String commitIdFirstCommit = featureBranchOne.getCommitIdFirstCommit();
+
+		FeatureBranch testCompare = featureBranchRepository.findByCommitIdThatTriggeredBuild("G011_TB");
+
+		assertEquals("Expected FeatureBranch Commit ID did not match ; findByCommitIdThatTriggeredBuild() not functioning as required ",
+			testCompare.getCommitIdFirstCommit(), commitIdFirstCommit);
+	} 	
+
+	@Test
+	public void testFindByCommitIdFirstCommit()
+	{
+		featureBranchRepository.save(featureBranchOne);
+		String commitIdThatTriggeredBuild = featureBranchOne.getCommitIdThatTriggeredBuild();
+
+		FeatureBranch testCompare = featureBranchRepository.findByCommitIdFirstCommit("G001_FC");
+
+		assertEquals("Expected FeatureBranch Commit ID did not match ; findByCommitIdFirstCommit() not functioning as required ",
+			testCompare.getCommitIdThatTriggeredBuild(), commitIdThatTriggeredBuild);		
+	}
+
+
+	/*
+		For tests, FeatureBranch objects are created in a manner such that
+			
+			featureBranchOne.firstCommitTimeStamp < featureBranchTwo.firstCommitTimeStamp  < featureBranchThree.firstCommitTimeStamp 
+			
+			featureBranchOne.deployTimeStamp < featureBranchTwo.deployTimeStamp < featureBranchThree.deployTimeStamp
+
+			FeatureBranch.firstCommitTimeStamp < FeatureBranch.deployTimeStamp
+	*/
+	
+	/*
+		For Functions findByTimeFrame(), findByFirstCommitTimeFrame() and findByDeployTimeFrame() 
+		of FeatureBranchRepository,
+
+		Case 1 : let timeStamp1 < fetureOne.firstCommitTimeStamp
+				 and timeStamp2 = featureBranchThree.deployTimeStamp  
+
+		Expectations:
+				 All 3 functions must return 3 FeatureBranch objects
+	*/
+
+	@Test
+	public void testCaseOne()
+	{
```

Comment:
Could we separate testing the three methods that you introduced into dedicated test cases for each?